#! /usr/bin/env python  
PKG = "robo50_node"
import roslib;roslib.load_manifest(PKG)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("update_rate", double_t, 0, "Polling rate for the robot.", 30.0, 5.0, 30.0 )

gen.add("cmd_vel_timeout", double_t, 0, "How long to wait before timing out on a velocity command..", 0.6, 0.0, 1.0)

gen.add("stop_motors_on_bump", bool_t, 0, "Stops motors when the bumps sensor is hit.", True)

gen.add("imu_used", bool_t, 0, "Enables or disables the imu.", True)

gen.add("move", bool_t, 0, "Enables or disables driving", True)

gen.add("gyro_scale_correction", double_t, 0, "Scaling factor for correct gyro operation.", 1.35, 0.0, 6.0)

gen.add("odom_angular_scale_correction", double_t, 0, "A correction applied to the computation of the rotation in the odometry.", 1.0, 0.0, 3.0)

gen.add("odom_linear_scale_correction", double_t, 0, "A correction applied to the computation of the translation in odometry.", 1.0, 0.0, 3.0)

gen.add("min_abs_yaw_vel", double_t, 0, "Minimum angular velocity of the TurtleBot.", None, 0.0, 3.0)
gen.add("max_abs_yaw_vel", double_t, 0, "Maximum angular velocity of the TurtleBot.", None, 0.0, 3.0)

exit( gen.generate(PKG, "Robo50", "Robo50"))
